### PR TITLE
Rework tests

### LINKED ISSUE #76: change assertions in test to unittest ones
as noted in #72 currently most assertions arent being done with the assertions of the unittest package. All assertions should get replaced by the assertions available in this package.

### PR DESCRIPTION
**Pull request by BlueBlueBlob**
## Description

--Your description here--

**Related issue (if applicable):** fixes #76 #56 #55 

## Checklist
  - [x] The code change is tested and works locally.
  - [x] tests pass. **Your PR won't be merged unless tests pass**
  - [x] There is no commented out code in this PR

### CODE REVIEW

**Comment by BlueBlueBlob** [comment]
Next commits will fix #56 #55

**Comment by BlueBlueBlob** [comment]
Need to fix some stuff and add execute_chore

File: test/test_grocy.py [hunk 1]
```diff
@@ -308,15 +354,21 @@ def test_add_product_error(self):
...
         self.assertIsNone(self.grocy.consume_product(1, 1.3, datetime.now()))
 
     @responses.activate
     def test_consume_product_error(self):
         responses.add(responses.POST, f"{CONST_BASE_URL}:{CONST_PORT}/api/stock/products/1/consume", status=400)
         self.assertRaises(HTTPError, self.grocy.consume_product, 1, 1.3, datetime.now())
+
+    @responses.activate
+    def test_execute_chore_valid(self):
+        date_exec_chore = datetime.now()
```
Comment (SebRut) [inline test/test_grocy.py]: This shouldn't be needed that way as the parameter is `datetime.now()` by default. Either change this to a fixed value (preferred) or remove this. Also valid for the remaining `datetime.now()` in the tests.

Comment (BlueBlueBlob) [reply test/test_grocy.py]: For coverage you need to pass all parameters even if they have a default value.
I will correct with a fixed value

File: test/test_grocy.py [hunk 2]
```diff
@@ -345,30 +345,29 @@ def test_get_last_db_changed_invalid_no_data(self):
     @responses.activate
     def test_add_product_valid(self):
         responses.add(responses.POST, f"{CONST_BASE_URL}:{CONST_PORT}/api/stock/products/1/add", status=200)
-        self.assertIsNone(self.grocy.add_product(1, 1.3, 2.44, datetime.now()))
+        self.assertIsNone(self.grocy.add_product(1, 1.3, 2.44, datetime.strptime("2019-05-04 11:31:04", '%Y-%m-%d %H:%M:%S')))
```
Comment (SebRut) [inline test/test_grocy.py]: I guess you could extract this into a global variable instead of creating a new instance all the time.

Comment (BlueBlueBlob) [reply test/test_grocy.py]: Sure. And the base url format too.

File: test/test_grocy.py [hunk 3]
```diff
@@ -17,6 +17,8 @@ def setUp(self):
         self.grocy = Grocy(CONST_BASE_URL, "api_key")
         self.grocy = None
         self.grocy = Grocy(CONST_BASE_URL, "demo_mode",  verify_ssl = CONST_SSL, port = CONST_PORT)
```
Comment (SebRut) [inline test/test_grocy.py]: I just saw this, but this line is overriding line 17

**Comment by BlueBlueBlob** [comment]
> remove double assignment of `self.grocy`

Coverage trick, to cover init Grocy.

File: test/test_grocy.py [hunk 4]
```diff
@@ -1,37 +1,78 @@
...
-        assert isinstance(chores, list)
-        self.assertEqual(len(chores), 6)
-        assert chores[0].id == 1
-        assert chores[1].id == 2
-        assert chores[2].id == 3
-        assert chores[3].id == 4
-        assert chores[4].id == 5
-        assert chores[5].id == 6
+        self.assertIsInstance(chores, list)
+        self.assertGreaterEqual(len(chores), 1)
```
Comment (SebRut) [inline test/test_grocy.py]: is it possible to change these into absolute values, switching from `GreaterEqual` to `Equal`...

Comment (BlueBlueBlob) [reply test/test_grocy.py]: Sure but it's not a mocking response.
Should I restore to a mocking test ?
For tests through real server I suggest "relative" (not sure about this word) tests.
Let me know.

Comment (SebRut) [reply test/test_grocy.py]: I thought we had control about the test data. If that's the case we should pin down the assertions as much as possible.

Comment (BlueBlueBlob) [reply test/test_grocy.py]: It's demo data, I don't know what exactly is in. But it covers most of case.
But it was the purpose of using a real Grocy instance ?

Comment (SebRut) [reply test/test_grocy.py]: Oh it's fine then.

**Comment by BlueBlueBlob** [comment]
Are all reviews good ?

**Pull request merged**